util/secret/certificates.go

```diff
@@ -189,27 +190,39 @@ func (c Certificates) GetByPurpose(purpose Purpose) *Certificate {
 }
 
 // Lookup looks up each certificate from secrets and populates the certificate with the secret data.
-func (c Certificates) Lookup(ctx context.Context, ctrlclient client.Client, clusterName client.ObjectKey) error {
+func (c Certificates) Lookup(ctx context.Context, secretCachingClient, ctrlclient client.Client, clusterName client.ObjectKey) error {
```

Comment:
We have to decide if we want to just make this breaking change or basically copy & paste this function and LookupOrGenerate.

I think usually nobody almost nobody is using this util and it's not too hard to adopt. But I'm fine with going through the normal deprecation

@fabriziopandini @killianmuldoon @ykakarap  opinions?

Reply:
I will go through deprecation, we are after the first beta date and we can make an alias without this paramater + handle this being null

Reply:
Hm the UX becomes a bit awkward when folks are supposed to pass in a nil client in some cases

Reply:
Pushed a new version, WDYT? (+/- needs more godoc if we go with that variant)